pkg/middlewares/compress/compresshandler/zstd_test.go

```diff
@@ -0,0 +1,864 @@
+package compresshandler
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/klauspost/compress/zstd"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	smallTestBody = []byte("aaabbc" + strings.Repeat("aaabbbccc", 9) + "aaabbbc")
+	bigTestBody   = []byte(strings.Repeat(strings.Repeat("aaabbbccc", 66)+" ", 6) + strings.Repeat("aaabbbccc", 66))
+)
+
+func Test_Zstandard_Vary(t *testing.T) {
+	h := newTestZstandardHandler(t, smallTestBody)
+
+	req, _ := http.NewRequest(http.MethodGet, "/whatever", nil)
+	req.Header.Set(acceptEncoding, "zstd")
+
+	rw := httptest.NewRecorder()
+	h.ServeHTTP(rw, req)
+
+	assert.Equal(t, http.StatusAccepted, rw.Code)
+	assert.Equal(t, acceptEncoding, rw.Header().Get(vary))
+}
+
+func Test_Zstandard_SmallBodyNoCompression(t *testing.T) {
+	h := newTestZstandardHandler(t, smallTestBody)
+
+	req, _ := http.NewRequest(http.MethodGet, "/whatever", nil)
+	req.Header.Set(acceptEncoding, "zstd")
+
+	rw := httptest.NewRecorder()
+	h.ServeHTTP(rw, req)
+
+	// With less than 1024 bytes the response should not be compressed.
+	assert.Equal(t, http.StatusAccepted, rw.Code)
+	assert.Empty(t, rw.Header().Get(contentEncoding))
+	assert.Equal(t, smallTestBody, rw.Body.Bytes())
+}
```

Comment:
```suggestion

func Test_Vary(t *testing.T) {
	testCases := []struct {
		desc           string
		h              http.Handler
		acceptEncoding string
	}{
		{
			desc:           "brotli",
			h:              newTestBrotliHandler(t, smallTestBody),
			acceptEncoding: "br",
		},
		{
			desc:           "zstd",
			h:              newTestZStandardHandler(t, smallTestBody),
			acceptEncoding: "zstd",
		},
	}

	for _, test := range testCases {
		test := test
		t.Run(test.desc, func(t *testing.T) {
			t.Parallel()

			req, _ := http.NewRequest(http.MethodGet, "/whatever", nil)
			req.Header.Set(acceptEncoding, test.acceptEncoding)

			rw := httptest.NewRecorder()
			test.h.ServeHTTP(rw, req)

			assert.Equal(t, http.StatusAccepted, rw.Code)
			assert.Equal(t, acceptEncoding, rw.Header().Get(vary))
		})
	}
}

func Test_SmallBodyNoCompression(t *testing.T) {
	testCases := []struct {
		desc           string
		h              http.Handler
		acceptEncoding string
	}{

		{
			desc:           "brotli",
			h:              newTestBrotliHandler(t, smallTestBody),
			acceptEncoding: "br",
		},
		{
			desc:           "zstd",
			h:              newTestZStandardHandler(t, smallTestBody),
			acceptEncoding: "zstd",
		},
	}

	for _, test := range testCases {
		test := test
		t.Run(test.desc, func(t *testing.T) {
			t.Parallel()
			h := newTestBrotliHandler(t, smallTestBody)

			req, _ := http.NewRequest(http.MethodGet, "/whatever", nil)
			req.Header.Set(acceptEncoding, test.acceptEncoding)

			rw := httptest.NewRecorder()
			h.ServeHTTP(rw, req)

			// With less than 1024 bytes the response should not be compressed.
			assert.Equal(t, http.StatusAccepted, rw.Code)
			assert.Empty(t, rw.Header().Get(contentEncoding))
			assert.Equal(t, smallTestBody, rw.Body.Bytes())
		})
	}
}
```

I will add a suggestion for each test but all the tests can written on this base.
`brotli_test.go` and `zstd_test` can be "merged" inside one file based on that.

Reply:
I thought the same, I'll update the main test suite to be iterative following your suggestion.